MediaTracker/app/src/main/java/com/izor066/android/mediatracker/ui/SearchGoogleBooks.java

```diff
@@ -0,0 +1,260 @@
+            inputMethodManager.hideSoftInputFromWindow(this.getCurrentFocus().getWindowToken(), 0);
+            return true;
+        }
+        return false;
+    }
+
+    @Override
+    public void onResultClick(Book book) {
+        Toast.makeText(this, "View details for: " + book.getTitle(), Toast.LENGTH_SHORT).show();
+        Intent intent = new Intent(this, SearchResultDetails.class);
+        intent.putExtra("Book", book);
+        this.startActivity(intent);
+    }
+
+    @Override
+    public void onResultAddClick(Book book) {
+        Toast.makeText(this, "Added entry: " + book.getTitle(), Toast.LENGTH_SHORT).show();
+        MediaTrackerApplication.getSharedDataSource().insertBookToDatabase(book);
+    }
+
+    private static class SearchTask extends AsyncTask<String, Void, Volumes> {
+        private final List<Book> resultsToAdd;
+        private final SearchResultsAdapter searchResultsAdapter;
+        private final ProgressBar pb;
+        private final RecyclerView recyclerView;
+        private final SimpleDateFormat simpleDateformatComplete = new SimpleDateFormat("yyyy-MM-dd");
+        private final SimpleDateFormat simpleDateFormatYm = new SimpleDateFormat("yyyy-MM");
+        private final SimpleDateFormat simpleDateFormatY = new SimpleDateFormat("yyyy");
+        private final String IMAGE_PLACEHOLDER = "https://s.gr-assets.com/assets/nophoto/book/blank-133x176-8b769f39ba6687a82d2eef30bdf46977.jpg";
+        private final String AUTHOR_PLACEHOLDER = "Anonymous";
+
+        private final String TAG = getClass().getSimpleName();
+
+        public SearchTask(List<Book> resultsToAdd, SearchResultsAdapter searchResultsAdapter, ProgressBar pb, RecyclerView recyclerView) {
+            this.resultsToAdd = resultsToAdd;
+            this.searchResultsAdapter = searchResultsAdapter;
+            this.pb = pb;
+            this.recyclerView = recyclerView;
+        }
+
+        @Override
+        protected void onPreExecute() {
+            resultsToAdd.clear();
+            searchResultsAdapter.notifyDataSetChanged();
+            pb.setVisibility(View.VISIBLE);
+        }
+
+        @Override
+        protected Volumes doInBackground(String... params) {
+            Log.v(TAG + "Search term", String.valueOf(params));
+            return Search.searchVolumes(params[0]);
+        }
+
+        @Override
+        protected void onPostExecute(Volumes searchListResponse) {
+            if (searchListResponse == null) {
+                pb.setVisibility(View.GONE);
+                return;
+            }
+
+            Log.d(TAG, String.valueOf(searchListResponse));
+
+
+            if (searchListResponse.getItems() == null) {
+                pb.setVisibility(View.GONE);
+                return;
+            }
+
+            for (int i = 0; i < searchListResponse.getItems().size(); i++) {
+                Volume volume = searchListResponse.getItems().get(i);
+
+                String authorsAll;
```

Comment:
I would suggest extracting some methods here. For example a method that returns the authors would be:

```java
// rest of the code from onPostExecute() .....

List<String> authors = volume.getVolumeInfo().getAuthors();
String authorsAll = createAllAuthorsFrom(authors);

// rest of the code ....

private String createAllAuthorsFrom(List<String> authors) {
    if (authors == null) {
        return AUTHOR_PLACEHOLDER;
    }

    StringBuilder sb = new StringBuilder();
    for (String author : authors) {
        sb.append(", " + author);    
    }
    return sb.toString().replaceFirst(", ","");
}
```

The same would go with `thumbnail` and `date` as their creation implies a little bit of logic, it is easier to read when you create separate methods for them.